r/R/matrix.R

```diff
@@ -158,6 +157,36 @@ all_matrix_inputs <- function(x) {
   x
 }
 
+#' Get the max of each row in an iterable matrix
+#' @param x IterableMatrix object/dgCMatrix object
+#' @export
+row_max <- function(x) {
```

Comment:
For naming clarity, I think we'll want to do a similar approach as `rowVars()`/`colVars()` below, namely:
1. Define a `rowMaxs` generic function with the same arguments as `matrixStats::rowMaxs`
2. Define `rowMaxs.default` to call out to `matrixStats` or `matrixGenerics` if present
3. Put our actual implementaiton in `rowMaxs.IterableMatrix`

This will make it so our users can use `rowMaxs()` mostly without worry. (If they call `matrixStats::rowMaxs()` directly it won't work for BPCells objects, but any other copy of `rowMaxs` they call will support the full range of matrix types)

Reply:
I think this makes interfacing with BPCells a lot more elegant, however for the future I wonder if doing this for each function will end up creating a lot of bloat.  Maybe we could create a function to automatically perform this strategy, given the default libraries with the same function names?

Reply:
That is a good idea! I'm not quite sure about how to programmatically handle defining functions with a given set of arguments, but it's probably something we could figure out with the Advanced R book.